Sources/SwiftParser/Parser.swift

```diff
@@ -237,6 +262,7 @@ public struct Parser {
 extension Parser {
   /// Retrieves the token following the current token without consuming it.
   func peek() -> Lexer.Lexeme {
+    lexemes.recordFurthestOffset()
```

Comment:
Ah, I didn't do that because I was trap into some weird bugs and fail the tests then. And it turned out that it was because I called `recordNextTokenInLookaheadTracker` in the wrong time🥹.

> Or is there a performance regression associated with that?
About the performance, see my comments below.